azure-core/azure/core/pipeline/policies/redirect.py

```diff
@@ -56,8 +56,8 @@ class RedirectPolicy(HTTPPolicy):
 
     def __init__(self, **kwargs):
         self.redirect_max = kwargs.pop('redirect_max', 30)
-        remove_headers = kwargs.pop('redirect_remove_headers', [])
-        self.remove_headers_on_redirect = set(remove_headers + self.REDIRECT_HEADERS_BLACKLIST)
+        remove_headers = kwargs.pop('redirect_remove_headers', set())
```

Comment:
Is `kwargs['redirect_remove_headers']` guaranteed to be `set`, when it does exist? If not, it might be even more defensive to do:
```
remove_headers = set(kwargs.pop('redirect_remove_headers', []))
```
Similar comment on other locations below.